### DISCUSSION TITLE
No buttons displaying in data importer - Direct link to CSV processing?

### DISCUSSION THREAD

**Discussion by robotictacos** [firefly-iii/firefly-iii#7956] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [ ] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I have firefly working with no issues.  Data importer does start up, but I'm not able to click anything to start processing my CSV.  Is there a direct link I can use to start chugging away at the CSV I downloaded?  Nordigen and Spectre don't seem to support US banks so it's the only option available to me.
![image](https://github.com/firefly-iii/firefly-iii/assets/47487583/3b8f9f5c-e009-469f-a0b3-e4af23f3e4dd)

### Debug information

Debug information generated at 2023-09-12 15:07:22 America/New_York for Firefly III Data Importer version **1.3.8**.

| Scope | Version |
| --- | --- |
| FIDI | 1.3.8 |
| Build | 317, 01-09-2023 18:06:24 UTC |
| PHP | 8.2.9 |
| Host | Linux |

| System info | Value |
| --- | --- |
| Using docker? | true |
| System TZ | America/New_York |
| Browser TZ | America/New_York |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | debug, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | apache2handler |
| BCscale | 12 |
| Trusted proxies (.env) |  |

| User info | Value |
| --- | --- |
| User agent | Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/117.0 |

### Expected behaviour

A way to start using the data importer 

### Steps to reproduce

1. Open home page for data importer

### Additional info

Here are the relevant docker compose sections:
###################################
###################################

  budget-app:
    image: fireflyiii/core:latest
    hostname: budget-app
    container_name: budget-app
    restart: always
    volumes:
      - $DOCKERDIR/firefly/storage:/var/www/html/storage/upload
    env_file: .budget.env
    depends_on:
      - budget-db
    networks:
      t2_proxy:
        ipv4_address: 172.18.0.18
    labels:
      - "traefik.enable=true"
      ## HTTP Routers
      - "traefik.http.routers.budget-rtr.entrypoints=https"
      - "traefik.http.routers.budget-rtr.rule=Host(`budget.$DOMAINNAME`)"
      ## Middlewares
      - "traefik.http.routers.budget-rtr.middlewares=chain-no-auth@file"
      ## HTTP Services
      - "traefik.http.routers.budget-rtr.service=budget-svc"
      - "traefik.http.services.budget-svc.loadbalancer.server.port=8080"

  budget-db:
    image: mariadb
    hostname: budget-db
    container_name: budget-db
    restart: always
    environment:
       MYSQL_ROOT_PASSWORD: 'supersecret'
       MYSQL_DATABASE: 'firefly'
       MYSQL_USER: 'firefly'
       MYSQL_PASSWORD: 'lesssecret'
    networks:
      t2_proxy:
        ipv4_address: 172.18.0.19
    volumes:
      - $DOCKERDIR/firefly/db:/var/lib/mysql
  
  budget-cron:
    image: alpine
    restart: always
    container_name: budget-cron
    command: sh -c "echo \"0 3 * * * wget -qO- http://budget-app:8080/api/v1/cron/pEDq17JlS+aqcmOSlevDala5rksC5FjL\" | crontab - && crond -f -L /dev/stdout"
    networks:
      t2_proxy:
        ipv4_address: 172.18.0.23

  budget-importer:
    image: fireflyiii/data-importer:latest
    hostname: budget-importer
    restart: always
    container_name: budget-importer
    networks:
      t2_proxy:
        ipv4_address: 172.18.0.24
    depends_on:
      - budget-app
    env_file: .budgetimporter.env
    labels:
      - "traefik.enable=true"
      ## HTTP Routers
      - "traefik.http.routers.budget-importer-rtr.entrypoints=https"
      - "traefik.http.routers.budget-importer-rtr.rule=Host(`budget-importer.$DOMAINNAME`)"
      ## Middlewares
      - "traefik.http.routers.budget-importer-rtr.middlewares=chain-no-auth@file"
      ## HTTP Services
      - "traefik.http.routers.budget-importer-rtr.service=budget-importer-svc"
      - "traefik.http.services.budget-importer-svc.loadbalancer.server.port=8080"

**Comment by JC5** [comment]
It looks like a reverse proxy configuration error. I see that the CSS isn't loaded and the JS probably isn't loaded either.

**Comment by lapo-gh** [comment]
I am experiencing the same issue with LSIO Swag reverse proxy. Any way to fix this?

**Reply by lapo-gh** [reply]
I managed to get it working - I added the environment variables "EXPECT_SECURE_URL=true" and the "VANITY_URL" and "FIREFLY_III_URL" both to the https://fireflyurl.local (replace with your own).
All working ok now.

**Reply by AdrianHordyk** [reply]
I had to set FIREFLY_III_URL to internal ip and port number, and VANITY_URL to https domain name to make it work

**Comment by simon-77** [comment]
Hi, I was also struggling to setup the reverse proxy (Traefik Proxy) for the importert.

Beside setting`FIREFLY_III_URL` and `VANITY_URL` (as explained in the [documentation](https://docs.firefly-iii.org/data-importer/installation/configuration/)) - I also had to set in the `.importer.env` file:
```
TRUSTED_PROXIES=*
EXPECT_SECURE_URL=true
```
I noticed the environment variable `TRUSTED_PROXIES` as described in the [Docs: Docker Installation](https://docs.firefly-iii.org/firefly-iii/installation/docker/#docker-and-reverse-proxies) - but I did not notice that the variable has to be set for the importer separately.

**Comment by simon-77** [comment]
While I was debugging the issue, I saw the error in my Browser Network Inspector: "Blocked loading mixed active content"

A quick internet search brought up this discussion at traefik forum: https://community.traefik.io/t/tls-termination-requests-work-but-put-responses-trigger-blocked-loading-mixed-active-content/16478/2
... coming to a conclusion that this is not an issue with the proxy itself.

@JC5 asking just in case you are using "Pythan and FastAPI" for the Data Importer tool:
The user in the discussion I linked above reported back an issue that "trailing slashes" should not be used.

Nevertheless, setting these two environment variables resolved the issue for me - thank you for this great tool.

**Reply by JC5** [reply]
I am not, no, but thanks for the tip!
